modules/util/stixhelpers.py

```diff
@@ -133,6 +133,50 @@ def get_technique_id_domain_map(ms):
     
     return tech_list
 
+def add_replace_or_ignore(obj_list, obj_in_question):
+    """ Add if object does not already exist
+        Replace object if exist depending on deprecation status or modified date
+        Ignore if object already exists but object in question is outdated
+    """
+
+    def get_object_in_conflict():
+        """ Return object if already in list. If there is a conflict with the STIX ID or ATT&CK ID """
+        for obj in obj_list:
+            if obj.id == obj_in_question.id or buildhelpers.get_attack_id(obj) == buildhelpers.get_attack_id(obj_in_question):
+                return obj
+        return []
+
+    # Get object in conflict
+    obj_in_conflict = get_object_in_conflict()
+
+    # Add: Object does not exist
+
+    # Verify if object does not appear with STIX ID or ATT&CK ID
+    if not obj_in_conflict:
+        obj_list.append(obj_in_question)
+
+    # Replace: Object already exists
+
+    # Ignore if object in question is deprecated and object in conflict is not
+    elif not obj_in_conflict.get('x_mitre_deprecated') and obj_in_question.get('x_mitre_deprecated'):
+        pass
+
+    # If object in conflict is deprecated and new object is not, select new
+    elif obj_in_conflict.get('x_mitre_deprecated') and not obj_in_question.get('x_mitre_deprecated'):
+        # Replace object in conflict with object in question
+        obj_list = [obj_in_question if x == obj_in_conflict else x for x in obj_list]
+
+    # Replace if modified date is newer
+    else:
+        conflict_modified = obj_in_conflict.get('modified')
+        in_question_modified = obj_in_question.get('modified')
+
+        if in_question_modified > conflict_modified:
+            # Replace object in conflict with object in question
+            obj_list = [obj_in_question if x == obj_in_conflict else x for x in obj_list]
```

Comment:
This would also be made more efficient by using a dict instead of a list since the list won't need to be rebuilt for every updated object (`O(1)` instead of `O(N)`), though that only occurs rarely so it's not a notable improvement. 

Plus that doesn't really change the overall efficiency since in Big-O efficiency you only consider the slowest part of the algorithm, which is already `O(N)` in the proposed improvement.